sdk/messaging/azservicebus/internal/utils/retrier.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
@jhendrixMSFT: This is just moved. I had to do a little bit of package splitting to make it so I could use the entityManager in the amqplinks_test.go (otherwise there were a few spots where circular references were happening).

This is basically the same as the alg that's in azcore right now for retry, minus HTTP status code checking. It's also internal for now, in case we decide to do some consolidation later.

Reply:
Presumably we'll need this, or something similar for EH?  Makes me think that we might want an `amqpcore` or something so that we can have a pipeline/policies that handle `*amqp.Message` types.

Reply:
I think that's a great idea. amqpLinks as well.

(although I'm waiting to do that until absolutely necessary since it'll slow down development of azservicebus at this moment).